src/mpi_collectives/include/mpi_collectives.h

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * Copyright (c) 2018 by Contributors
+ */
+
+#ifndef MXNET_MPI_COLLECTIVES_INCLUDE_MPI_COLLECTIVES_H_
+#define MXNET_MPI_COLLECTIVES_INCLUDE_MPI_COLLECTIVES_H_
+
+#if MXNET_USE_MPI_DIST_KVSTORE
+
+#include <mxnet/ndarray.h>
+
+#include <vector>
+#include <string>
+
+namespace mxnet {
+namespace kvstore {
+
```

Comment:
Although some relevant documentations are added to `c_api.h`, I still expect header only files like this come with detailed documentation for functionality description, arguments and return values.

Reply:
OK. I will add doxgen-style comment for these functions.